lib/rspec/core/world.rb

```diff
@@ -78,8 +78,7 @@ def exclusion_filter
       #
       # Get count of examples to be run.
       def example_count(groups=example_groups)
-        FlatMap.flat_map(groups) { |g| g.descendants }.
-          inject(0) { |a, e| a + e.filtered_examples.size }
+        FlatMap.flat_map(groups).inject(0) { |a, e| a + e.descendant_filtered_examples.size }
```

Comment:
`FlatMap.flat_map(groups)` returns `groups`, so this can be simplified to:

``` ruby
groups.inject(0) { |a, e| a + e.descendant_filtered_examples.size }
```

That said, I'm having a hard time understanding why this fixes anything.  The old code used `filtered_examples` but took care of calling it on every group by flat mapping over `descendants` first and calling `filtered_examples` on each.  Internally, `descendant_filtered_examples` does basically the same thing (in a slightly different order) but in a way that seems like it should be equivalent.